Disable network interface

DanielGuramulta opened the pull request:
Feature mentioned in issue: #362

dNechita commented on sdk/include/aditof/device_construction_data.h:
```diff
@@ -44,9 +44,11 @@ namespace aditof {
-    USB,      //!< connects to target via USB
+    LOCAL, //!< on the target
+    USB,   //!< connects to target via USB
+#ifdef HAS_NETWORK
```
API shouldn't change based on some build option.

dNechita commented on sdk/src/device_enumerator_ethernet.cpp:
```diff
@@ -30,15 +30,18 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
device_enumerator_ethernet.h/cpp could be excluded by cmake when building without Network support.

dNechita commented on sdk/src/device_factory.cpp:
```diff
@@ -45,9 +49,11 @@ DeviceFactory::buildDevice(const aditof::DeviceConstructionData &data) {
     }
+#ifdef HAS_NETWORK
     case DeviceType::ETHERNET: {
         return std::unique_ptr<DeviceInterface>(new EthernetDevice(data));
```
Should report at runtime that network support is disabled.

DanielGuramulta commented:
@dNechita I have addressed all the issues you mentioned and fixed them:
1) The api is no longer changed(`findDevices` method and `DeviceType` enum)
2) Moved the error message that should be seen at runtime in the systems implementation. How this works: when the user calls `getCameraAtIp(...)` and the sdk was built with `WITH_NETWORK=off` an error message will be logged telling the user that the network interface is disabled and to rebuild with the correct options enabled.

Thank you!
